Update gui.py

ItchyVision opened the pull request:
In this commit is what I changed to get this to work on macOS tested on Apple Silicon. I REM'd out the "app = QApplication(sys.argv + ["-platform", "windows:darkmode=2"])" and added to use cocoa. There may be a better way to code this so that it works with Windows and macOS but I am not much of a developer. 

Also added code to allow the window to be resized and added a scroll bar when necessary. 

With these changes, I can start torf via cmdl.

SavageCore commented:
Thanks for this, will come back to you with a test build once I've reworked it to remain compatible with other OSes!

ItchyVision commented:
Hey!

No problem! I was bored while RED was down and started hacking at the gui.py to help get it working on macOS. With my limited programing skills and ChatGPT, got it to work with macOS including making the window resizable and added a scroll bar to the window for smaller screens. Later added additional code so that it can auto detect the OS (Windows/macOS/Linux) and load appropriate code for each platform. Haven't tested other OS's yet but believe it should​ work. My latest commit is to try to fix special/accented characters when creating torrent files. This may only be macOS related. Found a bug that causes the created torrents files not to have the complete files when added to a BT client causing the torrent to be missing files when seeding. I don't have the issue when using qBittorrent to create the torrent file against the same dir. So my last commit has been tested yet but here was my criteria:

- Create test cases with:

- File names containing special characters (e.g., fiłęñáme.txt).
- Hidden files (.hiddenfile).
- Various Unicode encodings (e.g., accented characters).

- Run the application in batch mode and verify behavior.

Here is my fork that I was working with if you are interested in seeing my changes. All my changes have been limited to gui.py

https://github.com/SavageCore/torf-gui/compare/main...ItchyVision:torf-gui:main

Let me know if you have any question or would like me to test your build.

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Tuesday, November 26th, 2024 at 5:49 PM, Oliver Sayers ***@***.***> wrote:

> Thanks for this, will come back to you with a test build once I've reworked it to remain compatible with other OSes!
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/SavageCore/torf-gui/pull/23#issuecomment-2502203317), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BM6FVEVH4PSUTJDMADUXIKD2CUCKHAVCNFSM6AAAAABRZWJ2LSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMBSGIYDGMZRG4).
> You are receiving this because you authored the thread.Message ID: ***@***.***>
